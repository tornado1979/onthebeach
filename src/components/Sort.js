import React, { useState } from 'react';
import PropTypes from 'prop-types';

import reg from '../helpers/reg';

const Sort = (props) => {
  // default active option is the 1st
  const [active, setActive] = useState(0);


  /**
  * @desc handle click
  * @param string $symbol - option's symbol, e.g. Â£
  * @return jsx.element
  */
  function getSymbol(symbol){
    switch(symbol){
      case 'star':
        return (
           <i className="fa fa-star"></i>
        )
      case 'price':
        return (<div className="circle">
        <div className="price-symbol">
        </div>
      </div>)
      default:
         return <p className="alphabetical">{symbol}</p>
    }
  }

  /**
  * @desc handle click
  * @param object $event - clicked element, either the parent <div> or the child <p>
  * @return n/a
  */
  function handleClick(event){
    // get the selected option
    const selectedOption = event.target.classList.contains('option') ?
     event.target : event.target.parentElement;

    const activeOption = parseInt(selectedOption.id);

    props.handleSortClick(event);

    // update state
    setActive(activeOption);
  }

  let msg;
  return (
    <div className="sort">
      {props.options.map((sort, idx) => {
      msg = reg(sort.text);
      return (<div className={active === idx ? 'option active' : 'option'}
                   id={idx}
                   key={sort.text}
                   onClick={handleClick}>
                <p>{msg.first}</p>
                <p className="emphasized">{msg.second}</p>
                {getSymbol(sort.symbol)}
              </div>)
    })}
    </div>
  )
}

Sort.defaultProps  = {
  options: [{text: 'sort 1', symbol: 'symbol1'}, {text: 'sort 2', symbol: 'symbol2'}, {text: 'sort 3', symbol: 'symbol3'}]
}

Sort.propTypes = {
  options: PropTypes.array,
  handleSortClick: PropTypes.func.isRequired,
}

export default Sort;
