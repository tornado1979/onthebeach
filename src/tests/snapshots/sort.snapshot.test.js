import React from 'react';
import renderer from 'react-test-renderer';
import Sort from '../../components/Sort';

describe('Sort', () => {
  it('renders correctly', () => {
    const mockClick = jest.fn();
    const tree = renderer.create(
      <Sort handleSortClick={mockClick}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
