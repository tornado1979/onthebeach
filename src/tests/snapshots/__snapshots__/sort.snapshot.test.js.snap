// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort renders correctly 1`] = `
<div
  className="sort"
  data-testid="sort"
>
  <div
    className="option active"
    id={0}
    onClick={[Function]}
  >
    <p>
      sort 
    </p>
    <p
      className="emphasized"
    >
      1
    </p>
    <p
      className="alphabetical"
    >
      symbol 1
    </p>
  </div>
  <div
    className="option"
    id={1}
    onClick={[Function]}
  >
    <p>
      sort 
    </p>
    <p
      className="emphasized"
    >
      2
    </p>
    <p
      className="alphabetical"
    >
      symbol 2
    </p>
  </div>
  <div
    className="option"
    id={2}
    onClick={[Function]}
  >
    <p>
      sort 
    </p>
    <p
      className="emphasized"
    >
      3
    </p>
    <p
      className="alphabetical"
    >
      symbol 3
    </p>
  </div>
</div>
`;
